src/GitHub.VisualStudio/Base/EnsureLoggedInSection.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Reactive.Linq;
+using GitHub.Api;
+using GitHub.Extensions;
+using GitHub.Models;
+using GitHub.Services;
+using GitHub.UI;
+using GitHub.VisualStudio.Base;
+using Microsoft.TeamFoundation.MVVM;
+using System.Globalization;
+using GitHub.Primitives;
+using System.Threading.Tasks;
+
+namespace GitHub.VisualStudio.TeamExplorer.Sync
+{
+    public class EnsureLoggedInSection : TeamExplorerSectionBase
+    {
+        readonly IRepositoryHosts hosts;
+        readonly IVSServices vsServices;
+
+        public EnsureLoggedInSection(ISimpleApiClientFactory apiFactory, ITeamExplorerServiceHolder holder,
+            IConnectionManager cm, IRepositoryHosts hosts, IVSServices vsServices)
+            : base(apiFactory, holder, cm)
+        {
+            IsVisible = false;
+            this.hosts = hosts;
+            this.vsServices = vsServices;
+        }
+
+        public override void Initialize(IServiceProvider serviceProvider)
+        {
+            base.Initialize(serviceProvider);
+            CheckLogin().Forget();
+        }
+
+        protected override void RepoChanged()
+        {
+            base.RepoChanged();
+            CheckLogin().Forget();
```

Comment:
The callers don't really care when `CheckLogin` is done, since `CheckLogin` contains all the logic to show the notification information. It's a a fire-and-forget thing.